extensions/vscode/src/extension/VsCodeExtension.ts

```diff
@@ -371,16 +372,23 @@ export class VsCodeExtension {
       const openFiles = vscode.workspace.textDocuments;
       if (openFiles.length === 1) {
         // the count is amount of last open files
-        this.sidebar.webviewProtocol.request("setActiveFilePath", "", [PEAR_CONTINUE_VIEW_ID]);
+        this.sidebar.webviewProtocol.request("setActiveFilePath", undefined, [PEAR_CONTINUE_VIEW_ID]);
       }
     });
 
-    this.ide.onDidChangeActiveTextEditor((filepath) => {
-      this.core.invoke("didChangeActiveTextEditor", { filepath });
-      this.sidebar.webviewProtocol.request("setActiveFilePath", filepath, [PEAR_CONTINUE_VIEW_ID]);
-    });
+    context.subscriptions.push(
+      vscode.window.onDidChangeActiveTextEditor((event) => {
```

Comment:
did we have to change to use the vscode's onDidChangeActiveTextEditor? just for the !event return?